### ISSUE TITLE
Pulling the package into VyOS

### ISSUE THREAD

**Issue by dmbaturin** [fgp/vyatta-aiccu#1]
I'd like to pull it into VyOS. Is it ready for production now? Are you going to make backwards-incompatible changes any soon?

**Comment by dmbaturin** [comment]
Oh, nevermind, it's EdgeOS-specific. I'll have to fork and heavily modify it either way.

**Comment by fgp** [comment]
I'm not planning to do any big changes any time soon. I'd love to see this in VyOS! I must admist that EdgeOS is the only VyOS-based system I've ever used, so I don't have an idea how far EdgeOS has diverged from VyOS by now. I also found the documentation on how to extend the CLI somewhat, um, lacking, so I mostly just grepped around /opt/vyatta on my EdgeRouter to figure out how things are done.

So to summarize, it's quite likely that this is EdgeOS-specifc, but not due to a conscious decision but rather by accident. If you point out EdgeOS-isms I'm using, I'll see what I can do about them. I can't promise anything - this package was mostly a weekend hack, and spare time's a bit scarce right now - but I'll do what I can.

**Comment by dmbaturin** [comment]
Well, there are multiple problems that prevent it from being merged into either system actually. ;)
The most EdgeOS-specific thing is the firewall templates that call ubnt-fw. In both VyOS and EdgeOS firewall templates are never included into interfaces though, they are autogenerated.
The other thing, I don't get why would you remove the autoconf files. They are used throughout both EdgeOS and VyOS because they make things a lot easier than if you use custom install scripts, especially if you have C/C++ files there.

**Comment by fgp** [comment]
Hm, so regarding the firewalling stuff, are you suggesting that I can simply remove the "firewall" directory, and things will still work? I put them in there because I looked around EdgeOS's template-cfg directory, and the various interface types there did contain "firewall" directories. So I decided to simply copy one of those.

Regarding the autoconf stuff - I removed those because I don't include any sources that need to be compiled anyway.  My experience with autoconf has mostly been that it makes hard problems solvable, but simple problems quite hard. Since installing my package is a ratther trivial matter - copying some stuff into vyatta-cfg and vyatta's sbin directory - I figured that using a shell script would be easier.

I don't really get why using autoconf makes things simpler, unless there's a compilation step invovled. But if it has tangible benefits, I'd be happy to switch.
